### PR TITLE
Slack notifications & syntax check

### PR DESCRIPTION
**Pull request by elblivion**
Adds Slack notification function. To make this work you will need to set SLACK_WEBHOOK in your CI env.

### CODE REVIEW

File: s3_package_deploy.sh [hunk 1]
```diff
@@ -2,6 +2,36 @@
...
+export SLACK_MSG_TITLE="deployment pipeline event"
+
+notify_slack() {
+
+  if [ -z $SLACK_WEBHOOK ]; then
+    echo "SLACK_WEBHOOK not configured"
+    return
+  fi
+
+  SLACK_EMOJI=${SLACK_EMOJI:-stopwatch}
```
Comment (danihodovic) [inline s3_package_deploy.sh]: Why are two slack variables exported while this is not?

**Comment by danihodovic** [comment]
No tests :trollface: ?

File: s3_package_deploy.sh (hunk 1 above)
Comment (elblivion) [inline s3_package_deploy.sh]: Actually I think the export is unnecessary because everything is probably in the same scope, but :shrug:

**Comment by elblivion** [comment]
Integration tests? :trollface:

File: s3_package_deploy.sh [hunk 2]
```diff
@@ -2,6 +2,36 @@
...
+     \"color\": \"$SLACK_MSG_COLOUR\", \
+     \"fallback\": \"\", \
+     \"title\": \"$SLACK_MSG_TITLE\", \
+     \"text\": \"${message}\", \
+     \"mrkdwn_in\": [\"text\"] \
+    } \
+  ] \
+}"
+  echo $payload
+  curl -sH "Content-Type: application/json" -X POST -d "${payload}" ${SLACK_WEBHOOK}
```
Comment (elblivion) [inline s3_package_deploy.sh]: Should probably check curl is installed.

**Comment by danihodovic** [comment]
Once the export variables is consistent, LGTM

**Comment by elblivion** [comment]
Also not actually using the notification anywhere!

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+test:
+  override:
+    - for script in *.sh; do bash -n $script; done
```
Comment (danihodovic) [inline circle.yml]: Are you really too lazy to make the scripts explicit :trollface:

Comment (elblivion) [inline circle.yml]: Yeah :-D

**Pull request merged**